internal/extsvc/azuredevops/client.go

```diff
@@ -126,14 +119,38 @@ func (c *Client) do(ctx context.Context, req *http.Request, result any) (*http.R
 	return resp, json.Unmarshal(bs, result)
 }
 
+// WithAuthenticator returns a new Client that uses the same configuration,
+// HTTPClient, and RateLimiter as the current Client, except authenticated with
+// the given authenticator instance.
+//
+// Note that using an unsupported Authenticator implementation may result in
+// unexpected behaviour, or (more likely) errors. At present, only BasicAuth is
+// supported.
+func (c *Client) WithAuthenticator(a auth.Authenticator) (*Client, error) {
+	switch a.(type) {
+	case *auth.BasicAuth:
+		break
+	default:
+		return nil, errors.Errorf("authenticator type unsupported for Azure DevOps clients: %s", a)
+	}
```

Comment:
Won't this work? (Haven't checked, but was surprised to find a switch statement with one `case` and one `default`)

```suggestion
	if _, ok := a.(*auth.BasicAuth); !ok {
		return nil, errors.Errorf("authenticator type unsupported for Azure DevOps clients: %s", a)
	}
```

Reply:
Originally I thought maybe it would be fine to leave it as a switch like the other WithAuthenticator methods in case we add more types in the future, but there's no real point to have it like that for now, so I'll change it to the if :).